akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala

```diff
@@ -219,6 +219,47 @@ class SourceSpec extends AkkaSpec {
     }
   }
 
+  "Unfold Source" must {
+    val expected = List(9227465, 5702887, 3524578, 2178309, 1346269, 832040, 514229, 317811, 196418, 121393, 75025, 46368, 28657, 17711, 10946, 6765, 4181, 2584, 1597, 987, 610, 377, 233, 144, 89, 55, 34, 21, 13, 8, 5, 3, 2, 1, 1, 0)
+
+    "generate a finite fibonacci sequence" in {
+      import GraphDSL.Implicits._
+
+      val source = Source.unfold(0 → 1) {
```

Comment:
consider the line: `case (a, b) ⇒ (b → (a + b)) → a` with the regular tuple syntax, it would look like: `case (a, b) ⇒ ((b,(a+b)),a)`, the arrow syntax saves parenthesis in an already heavy parenthesis expression. for the rest of the test, it's more a matter of uniformity. In any case, if you still think I should change the code to use the regular tuple syntax, I will change it.